host.go

```diff
@@ -518,6 +518,12 @@ func httpRequest(ctx context.Context, m api.Module, requestOffset uint64, bodyOf
 			panic(fmt.Errorf("invalid http request: %v", err))
 		}
 
+		// default method to GET and force to be upper
+		if request.Method == "" {
+			request.Method = "GET"
+		}
+		request.Method = strings.ToUpper(request.Method)
```

Comment:
as discussed in chat. we should do this to match the behavior of js-sdk. also, we uppercase because some http servers or APIs seem to be sensitive to this.